frontend/screens/Influencer/components/UserProfilePhoto.js

```diff
@@ -1,20 +1,101 @@
-import React from "react";
-import { StyleSheet, View, Text, TouchableOpacity } from "react-native";
+import React, { useState } from "react";
+import {
+  StyleSheet,
+  View,
+  Text,
+  TouchableOpacity,
+  Alert,
+  Platform,
+  ScrollView,
+} from "react-native";
 import { Image } from "expo-image";
+import * as ImagePicker from "expo-image-picker";
 import { UserProfilePhotoStyles } from "./UserProfilePhoto.scss";
+import { useAlert } from "../../../util/AlertContext";
 
-const UserProfilePhoto = ({route,navigation}) => {
+const UserProfilePhoto = ({ route, navigation }) => {
+  const [selectedImage, setSelectedImage] = useState(null);
   const social = route.params?.social;
+  const {showAlert} = useAlert()
   const follower = route.params?.follower;
   const price = route.params?.price;
-  const handleUploadPhoto = () => {
-    // Handle the upload photo action
-    console.log("Upload photo clicked");
+  const [photo, setPhoto] = useState(null);
+  const requestPermission = async () => {
+    if (Platform.OS !== "web") {
+      const { status } =
+        await ImagePicker.requestMediaLibraryPermissionsAsync();
+      if (status !== "granted") {
+        showAlert("Alert","Permission to access media library is required!");
+        return false;
+      }
+    }
+    return true;
   };
 
-  const handleTakePhoto = () => {
-    // Handle the take photo action
-    console.log("Take photo clicked");
+  const handleUploadPhoto = async () => {
+    const permission = await requestPermission();
+    if (!permission) return;
+
+    const result = await ImagePicker.launchImageLibraryAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.Images,
+      allowsEditing: true,
+      aspect: [1, 1],
+      quality: 1,
+    });
+
+    if (!result.canceled) {
+      setSelectedImage(result.assets[0].uri);
+      let localUri = result.assets[0].uri;
+      let filename = localUri.split('/').pop();
+      let match = /\.(\w+)$/.exec(filename);
+      let type = match ? `image/${match[1]}` : `image`;
+      setPhoto({
+        uri: localUri,
+        name: filename,
+        type: type
+      })
+    }
+  };
+
+  const handleTakePhoto = async () => {
+    if (Platform.OS === "web") {
+      document.getElementById("fileInput").click();
+      return;
+    }
+
+    const cameraPermission = await ImagePicker.requestCameraPermissionsAsync();
+    const mediaLibraryPermission = await requestPermission();
+
+    if (!cameraPermission.granted || !mediaLibraryPermission) {
+      showAlert("Alert","Permission to access camera and media library is required!");
+      return;
+    }
+
+    const result = await ImagePicker.launchCameraAsync({
+      allowsEditing: true,
+      aspect: [1, 1],
+      quality: 1,
+    });
+
+    if (!result.canceled) {
+      let localUri = result.assets[0].uri;
```

Comment:
same comment as above, can we expose a method in util folder and reuse instead of writing the same code again and again?

Reply:
No both are different. One is for upload from device. One is take picture from camera.

Reply:
There is one more occurence - https://github.com/Kajol-Kumari/InfluMart/pull/84/files#diff-20d169099eefc6163330d7c3e1a2d4f8f1773eb2dca4b0d843a0e39e6caecd8fR48 
I still feel atleast common parts can be seggregated

Reply:
Ok done. I will update it.